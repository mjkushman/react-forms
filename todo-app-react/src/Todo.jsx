const Todo = ({task, removeTodo}) => {
    
    const handleRemove = () =>{
        removeTodo()
    }
    
    return (
        <li>{task}<button onClick={handleRemove}>X</button></li>
    )
}

export default Todo