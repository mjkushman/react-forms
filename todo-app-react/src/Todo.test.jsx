import React from 'react'
import { render } from '@testing-library/react';
import {describe, test, expect} from 'vitest';
import Todo from './Todo.jsx'


// Smoke test
test('Renders without going up in smoke', ()=> {
  render(<Todo />)
})

// snapshot test
test("snapshot match", ()=> {
  const {asFragment} = render(<Todo/>);

  expect(asFragment()).toMatchSnapshot();
})