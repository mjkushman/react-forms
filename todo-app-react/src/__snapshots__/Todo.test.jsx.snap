// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot match 1`] = `
<DocumentFragment>
  <li>
    <button>
      X
    </button>
  </li>
</DocumentFragment>
`;
